### PR TITLE
Add Microsoft Azure support

### PR DESCRIPTION
**Pull request by ValchukDmitry**

### CODE REVIEW

File: schema/src/main/kotlin/io/kotless/KotlessConfig.kt [hunk 1]
```diff
@@ -11,13 +11,17 @@ import java.io.File
...
+ * @param cloud type of the cloud
  */
 data class KotlessConfig(
     val bucket: String,
     val prefix: String,
     val dsl: DSL,
     val terraform: Terraform,
-    val optimization: Optimization = Optimization()
+    val optimization: Optimization = Optimization(),
+    val cloud: String = "aws",
```
Comment (TanVD) [inline schema/src/main/kotlin/io/kotless/KotlessConfig.kt:22]: Let's make it enum class

File: schema/src/main/kotlin/io/kotless/KotlessConfig.kt [hunk 2]
```diff
@@ -11,13 +11,17 @@ import java.io.File
...
+    val resourceGroup: String?,
```
Comment (TanVD) [inline schema/src/main/kotlin/io/kotless/KotlessConfig.kt:23]: Probably should be moved to some abstraction for specific cloud. Like `interface CloudConfig` with implementation `AzureConfig`

File: engine/src/main/kotlin/io/kotless/gen/factory/azure/route53/RecordFactory.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    override fun mayRun(entity: Application.Route53, context: GenerationContext) =
+        context.output.check(entity, ZoneFactory)
+            && context.schema.lambdas.all.all { context.output.check(it, FunctionFactory) }
+            && context.output.check(context.webapp, InfoFactory)
+
+    override fun generate(entity: Application.Route53, context: GenerationContext): GenerationFactory.GenerationResult<Output> {
+        val lambda = context.output.get(context.schema.lambdas.all.first(), FunctionFactory)
+        val dnsZone = context.output.get(entity, ZoneFactory)
+//        val certificate = context.output.get(entity, CertificateFactory).certificate
```
Comment (TanVD) [inline engine/src/main/kotlin/io/kotless/gen/factory/azure/route53/RecordFactory.kt:24]: What about certificate?)

File: engine/src/main/kotlin/io/kotless/gen/AzureGenerator.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        Application.ApiGateway.StaticRoute::class to setOf(StaticRouteFactory),
+        Application.ApiGateway.DynamicRoute::class to setOf(DynamicRouteFactory)
+    )
+
+    fun generate(schema: Schema): Set<TFFile> {
+        val context = GenerationContext(schema, schema.application)
+
+        var newExecuted = true
+        while (newExecuted) {
+            println("---------------------------------")
```
Comment (TanVD) [inline engine/src/main/kotlin/io/kotless/gen/AzureGenerator.kt:41]: Hm?) Probably we should at least move it to tracing level?

File: dsl/kotless/kotless-lang/src/main/kotlin/io/kotless/dsl/FunctionHandler.kt [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+import io.kotless.InternalAPI
+import io.kotless.dsl.app.http.RouteKey
+import io.kotless.dsl.app.http.RoutesDispatcher
+import io.kotless.dsl.model.HttpRequest
+import io.kotless.dsl.utils.Json
+import org.slf4j.LoggerFactory
+import java.util.*
+
+@InternalAPI
+class FunctionHandler {
```
Comment (TanVD) [inline dsl/kotless/kotless-lang/src/main/kotlin/io/kotless/dsl/FunctionHandler.kt:20]: What about other frameworks?

File: dsl/kotless/kotless-lang/build.gradle.kts [hunk 6]
```diff
@@ -7,6 +7,7 @@ version = rootProject.version
 
 dependencies {
     api(project(":dsl:common:lang-common"))
+    implementation("com.microsoft.azure.functions", "azure-functions-java-library", "1.2.2")
```
Comment (TanVD) [inline dsl/kotless/kotless-lang/build.gradle.kts:10]: Not sure that the best way is to bundle Azure right into dsl artifact... Maybe we should split them into `aws` and `azure` part?

**Review by TanVD — CHANGES_REQUESTED** [review]

File: schema/src/main/kotlin/io/kotless/KotlessConfig.kt (hunk 1 above)
Comment (ValchukDmitry) [reply schema/src/main/kotlin/io/kotless/KotlessConfig.kt]: Fixed

File: dsl/kotless/kotless-lang/src/main/kotlin/io/kotless/dsl/FunctionHandler.kt (hunk 5 above)
Comment (ValchukDmitry) [reply dsl/kotless/kotless-lang/src/main/kotlin/io/kotless/dsl/FunctionHandler.kt]: I've added Ktor support

File: dsl/kotless/kotless-lang/build.gradle.kts (hunk 6 above)
Comment (ValchukDmitry) [reply dsl/kotless/kotless-lang/build.gradle.kts]: I tried to split these parts but looks like some cloud-specific libraries still include into artifacts

**Review by TanVD — APPROVED** [review]

**Pull request merged**